module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -165,33 +160,133 @@ namespace module::input {
+                            auto& original = utility::platform::getRawServo(i, *s);
+                            // Add the sensor values to the system properly
+                            filtered_sensors->servo.push_back({0,
+                                                               i,
+                                                               original.torque_enabled,
+                                                               original.p_gain,
+                                                               original.i_gain,
+                                                               original.d_gain,
+                                                               original.goal_position,
+                                                               original.moving_speed,
+                                                               original.present_position,
+                                                               original.present_speed,
+                                                               original.load,
+                                                               original.voltage,
+                                                               float(original.temperature)});
+                        }
+
+                        // Calculate forward kinematics
+                        const auto Htx = calculateAllPositions(model, *filtered_sensors);
+                        for (const auto& entry : Htx) {
+                            filtered_sensors->Htx[entry.first] = entry.second.matrix();
+                        }
+
+                        // Calculate the average length of both legs from the torso and accumulate this measurement
+                        const Eigen::Affine3d Htr(filtered_sensors->Htx[ServoID::R_ANKLE_ROLL]);
+                        const Eigen::Affine3d Htl(filtered_sensors->Htx[ServoID::L_ANKLE_ROLL]);
+                        const Eigen::Vector3d rTFt = (Htr.translation() + Htl.translation()) * 0.5;
+
+                        // Accumulator CoM readings
+                        rMFt += calculateCentreOfMass(model, filtered_sensors->Htx).head<3>() + rTFt;
+                    }
+
+                    // Average all accumulated readings
+                    acc /= double(sensors.size());
+                    gyro /= double(sensors.size());
+                    rMFt /= double(sensors.size());
+
+                    // Average time per sensor reading
+                    double deltaT = std::chrono::duration_cast<std::chrono::duration<double>>(
+                                        sensors.back()->timestamp - sensors.front()->timestamp)
+                                        .count()
+                                    / double(sensors.size());
+
+                    // Find the rotation from the average accelerometer reading to world UnitZ
+                    // Rotating from torso acceleration vector to world z vector ===> torso to world rotation
```

Comment:
```
                    // Rotating from torso acceleration vector to world z vector ===> torso to world rotation
```
I don't know what this means. The arrow is throwing me off. Is that an "implies" arrow? `$\Longrightarrow$` vibe? it seems like this could be more of an `$\iff$` vibe

Reply:
`// we are performing this operation ====> therefore, this is the operation` (i.e. my logic behind me calling it `Rwt` and not `Rtw`)

Reply:
Could you please say that with words, and remove the ambiguity